tests/manage/pv_services/test_delete_project_with_pvc_and_pods.py

```diff
@@ -0,0 +1,165 @@
+"""
+The purpose of this test case is to delete at least 2 projects:
+One project with just pvcs and the other with pvcs attached to a pod
+"""
+import logging
+import random
+import time
+import pytest
+
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour
+from ocs_ci.framework.testlib import ManageTest, tier1
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.ocs.resources import pod
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.utility.retry import retry
+from tests import helpers
+
+log = logging.getLogger(__name__)
+
+
+@retry(UnexpectedBehaviour, tries=10, delay=3, backoff=1)
+def check_ceph_used_space():
+    """
+    Check for the used space in cluster
+    """
+    ct_pod = pod.get_ceph_tools_pod()
+    ceph_status = ct_pod.exec_ceph_cmd(ceph_cmd="ceph status")
+    assert ceph_status is not None
+    used = ceph_status['pgmap']['bytes_used']
+    used_in_gb = used / constants.GB
+    global used_space
+    if used_space and used_space == used_in_gb:
+        return used_in_gb
+    used_space = used_in_gb
+    raise UnexpectedBehaviour(
+        f"In Ceph status, used size is keeping varying"
+    )
+
+@tier1
+@pytest.mark.polarion_id("OCS-278")
+class TestDeleteProjectWithPVCAndPods(ManageTest):
+    """
+    Create at least 2 projects, one with just pvcs and the other with pvcs
+    attached to pod, and delete them.
+    """
+    cephfs_pvcs_num = 100
+    rbd_pvcs_num = 100
+
+    def test_delete_project_with_pvc_and_pods(self, project_factory,
+                                              storageclass_factory, pvc_factory,
+                                              multi_pvc_factory, pod_factory):
+        # Start with Project 1
+        log.info("Creating OCP+OCS Setup")
+        ocp_setup = ocp.OCP()
+        ocs_setup = OCS()  # TODO: Scale the setup to at-least 800 pvcs/pods
+
+        log.info("Creating CephFS Storage class")
+        cephfs_sc = storageclass_factory(interface=constants.CEPHFILESYSTEM)
+        log.info("Creating RBD Storage class")
+        rbd_sc = storageclass_factory(interface=constants.CEPHBLOCKPOOL)
+
+        log.info("Creating Project 1")
+        project_1 = project_factory()
+
+        log.info("Creating {} CephFS PVCs".format(self.cephfs_pvcs_num))
+        # Generate a given number of CephFS PVCs, randomly assigned RWO or RWX
+        cephfs_pvcs = [pvc_factory(interface=constants.CEPHFILESYSTEM,
+                                   project=project_1,
+                                   storageclass=cephfs_sc,
+                                   access_mode=random.choice(
```

Comment:
Instead of this random call, randomize the number of PVs between 0 and self.cephfs_pvcs_num that will use RWO, and the rest will use RWX. it's a single call to random.choice().

Reply:
OK - I will implement it as a single call outside of the list comprehension